cajita/src/Cajita_Parallel.hpp

```diff
@@ -0,0 +1,325 @@
+}
+
+//---------------------------------------------------------------------------//
+/*!
+  \brief Execute a functor in parallel with a multidimensional execution
+  policy specified by the given local grid, decomposition, and entity
+  type. The loop indices are local.
+
+  \tparam FunctorType The functor type to execute.
+
+  \tparam ExecutionSpace The execution space type.
+
+  \tparam MeshType The mesh type of the local grid.
+
+  \param label Parallel region label.
+
+  \param exec_space An execution space instance.
+
+  \param decomposition The decomposition type of the entities (own,ghost).
+
+  \param entity_type The entity type over which to loop.
+
+  \param functor The functor to execute.
+ */
+template <class FunctorType, class ExecutionSpace, class MeshType,
+          class DecompositionType, class EntityType>
+inline void
+grid_parallel_for( const std::string &label, const ExecutionSpace &exec_space,
+                   const LocalGrid<MeshType> &local_grid,
+                   const DecompositionType &decomposition,
+                   const EntityType &entity_type, const FunctorType &functor )
+{
+    auto index_space =
+        local_grid.indexSpace( decomposition, entity_type, Local() );
+    grid_parallel_for( label, exec_space, index_space, functor );
+}
+
+//---------------------------------------------------------------------------//
+/*!
+  \brief Execute a functor with a work tag in parallel with a multidimensional
+  execution policy specified by the given local grid, decomposition, and entity
+  type. The loop indices are local.
+
+  \tparam FunctorType The functor type to execute.
+
+  \tparam WorkTag The functor work tag.
+
+  \tparam ExecutionSpace The execution space type.
+
+  \tparam MeshType The mesh type of the local grid.
+
+  \param label Parallel region label.
+
+  \param exec_space An execution space instance.
+
+  \param decomposition The decomposition type of the entities (own,ghost).
+
+  \param entity_type The entity type over which to loop.
+
+  \param functor The functor to execute.
+ */
+template <class FunctorType, class WorkTag, class ExecutionSpace,
+          class MeshType, class DecompositionType, class EntityType>
+inline void
+grid_parallel_for( const std::string &label, const ExecutionSpace &exec_space,
+                   const LocalGrid<MeshType> &local_grid,
+                   const DecompositionType &decomposition,
+                   const EntityType &entity_type, const WorkTag &work_tag,
+                   const FunctorType &functor )
+{
+    auto index_space =
+        local_grid.indexSpace( decomposition, entity_type, Local() );
+    grid_parallel_for( label, exec_space, index_space, work_tag, functor );
+}
+
+//---------------------------------------------------------------------------//
+// Grid Parallel Reduce
+//---------------------------------------------------------------------------//
+/*!
+  \brief Execute a functor in parallel with a multidimensional execution
```

Comment:
Should probably add "reduce" to these brief descriptions

Reply:
good idea - added this